### ISSUE TITLE
Feature Request:  Allow mapping of data elements to the XDM object within the Action Component within Rules.

### ISSUE THREAD

**Issue by RudiShumpert** [adobe/alloy#552]
### Expected Behaviour

Within Launch, a user should be able to map a data element to an XDM node within the Rule Action Component of the XDM / AEP SDK extension.  Not only will this same time, but it will make migration to alloy much easier and lead to faster adoption.  This is the functionality we have now with the Adobe Analytics Extension.

A user should also be able to set global variables in the AEP web SDK extension configuration screen as well like the Analytics extension does now as well.

**Comment by jfkhoury** [comment]
@RudiShumpert Any chance you can attach some screenshots of what we have today in Analytics?

**Comment by RudiShumpert** [comment]
This screenshot shows the extension configuration screen for Analytics.  Note how you can assign global variables 
![Screenshot 2020-07-15 15 42 17](https://user-images.githubusercontent.com/37945125/87588616-f3cc8780-c6b1-11ea-8825-070fbeb8cb2e.png)

This screenshot show the rule action component to set variables in analytics.  You can add / set as many evars/props as you want to set and do the mapping in the rule. 
![Screenshot 2020-07-15 15 42 41](https://user-images.githubusercontent.com/37945125/87588628-f7f8a500-c6b1-11ea-8af4-0c2e52517bb4.png)

**Comment by ericmatisoff** [comment]
Adding a Set Variables Rule Action to the AEP Web SDK would be a huge improvement. It'd mean many less steps. 

As it lives today, 3 data elements are necessary to send most Web SDK events (whereas previously it'd have been just 1). 
Those three data elements:
1 - extracting the information from the data layer (or DOM)
2 - adding it to an XDM object
3 - combining multiple XDM objects into a final data element to be used in the Send Event.

**Comment by jonsnyder** [comment]
In AppMeasurement and the Analytics extension you set variables directly on the tracker.  The tracker holds the state of all the variables. Alloy doesn't have this. Every time you want to send an event you need to provide a data element. We could implement what you are describing though in this way: instead of setting the variables directly on the tracker, the action would set the variables on a data element. 

We built the XDM data element as a way to build an object, but what you are describing is modifying an object. We could create a "Modify XDM Data Element" action where you would select a data element, and then have a similar interface to the current XDM data element. The difference would be that you aren't creating it from scratch, the action simply modifies some of the fields.

Have I captured what you are describing?

**Comment by RudiShumpert** [comment]
@jonsnyder Would the Modify XDM object be within the rule action component ?

**Comment by jonsnyder** [comment]
@RudiShumpert, yes that's what I'm proposing.  The Modify XDM object would be a rule action.

**Comment by ericmatisoff** [comment]
One more use case for the ability to modify XDM Object data elements on the fly in a rule: 

A lot of customers will use Custom Code in a Rule's Condition in order to capture information about the DOM element that is being clicked. They'll do this in a matter similar to this: [https://share.getcloudapp.com/jkuZdjAE](https://share.getcloudapp.com/jkuZdjAE).

Then pull that data element (tempURL) into a field within the event beacon.

Without XDM Object editing in rules, these customers would have to save the rule, then switch to the XDM Object Data Element to add %tempURL% to the correct field. The workflow is much smoother today for Analytics since all of this can be accomplished in the rule.

**Comment by ericmatisoff** [comment]
Another painful use case today is that it takes several extra steps to debug issues found in a beacon:

1. See Issue in the beacon
2. Read console logs and see fired rule(s)
3. Check each rule’s Web SDK action to find the XDM Object being sent in the rule
4. Open the XDM Object data element to identify the issue. (Or, if you’re like me, and need to merge multiple XDM Objects together in a rule, open the merged data element of xdm objects that was reference in the rule, then:
5. Go to each of the merged data elements to find which one is setting the offending variable(s)

Instead, if most variables are set in rule actions instead of data element, the debugging would (possibly) end at Step 3.

**Comment by jfkhoury** [comment]
Jira ticket: https://jira.corp.adobe.com/browse/CORE-51889
